src/runtime/link-accounts-flow.js

```diff
@@ -306,44 +317,27 @@ export class LinkSaveFlow {
       }
     });
 
-    this.linkedPromise_ = this.activityIframeView_.port().then(port => {
+    this.openPromise_ = this.dialogManager_.openView(this.activityIframeView_,
+        /* hidden */ true);
+    /** {!Promise<boolean>} */
+    return this.activityIframeView_.port().then(port => {
       return acceptPortResultData(
           port,
           feOrigin(),
           /* requireOriginVerified */ true,
           /* requireSecureChannel */ true);
-    });
-
-    let linkConfirm = null;
-    this.confirmPromise_ = this.linkedPromise_.then(result => {
-      // This flow is complete
-      this.complete_();
-      if (result['linked']) {
-        this.dialogManager_.popupClosed();
-        this.deps_.callbacks().triggerFlowStarted(
-            SubscriptionFlows.LINK_ACCOUNT);
-        linkConfirm = new LinkCompleteFlow(this.deps_, result);
-        return linkConfirm.start();
-      }
-      return Promise.reject(createCancelError(this.win_, 'not linked'));
-    }).then(() => {
-      this.deps_.callbacks().triggerLinkProgress();
-      return linkConfirm.whenComplete();
-    }).then(() => {
-      return true;
+    }).then(result => {
+      return this.handleLinkSaveResponse_(result);
     }).catch(reason => {
+      // In case this flow wasn't complete, complete it here
+      this.complete_();
+      // Handle cancellation from user, link confirm start or completion here
       if (isCancelError(reason)) {
         this.deps_.callbacks().triggerFlowCanceled(
             SubscriptionFlows.LINK_ACCOUNT);
         return false;
       }
-      // In case this flow wasn't complete, complete it here
-      this.complete_();
       throw reason;
```

Comment:
if the error is not a cancel error - then wouldn't you want to return false after throwing a reason?

Reply:
We still have to complete the existing flow with the dialog manager. I actually moved the complete above isCancelError..mabye need to push the new patch.